src/Data/Nat/Divisibility/Core.agda

```diff
@@ -35,7 +35,27 @@ record _∣_ (m n : ℕ) : Set where
   constructor divides
   field quotient : ℕ
         equality : n ≡ quotient * m
-open _∣_ using (quotient) public
+
+  quotient≢0 : .{{NonZero n}} → NonZero quotient
```

Comment:
This could be declared as an instance to avoid having to use it manually?

Reply:
Well, that's how I had it originally, but @Taneb [pointed out](https://github.com/agda/agda-stdlib/pull/2182#discussion_r1371486070) this might give rise to the problems discussed in #1565 !? But you're the expert on `instance`s form this perspective, so I'd take your cue. But if we *do* want it as an instance, then shouldn't it be automatically exported by any instance of `_|_`, ie it should be part of the record API, not the 'properties'?

Again, I don't feel entirely on safe ground here as regards my intuitions, so greater experience/wisdom on such points is welcome!

Reply:
Hmm it's a good point. The case when the same issue would trigger is when `quotient` would compute to something of the form `suc n`... However, how likely is that? I think you'd have to have a concrete proof of divisibility for a concrete pair of numbers, e.g. you open an actual computed proof of `8 | 2` and you then try and pass `quotient = 4` as an argument to something that needs instance search.

Regardless, instances always trigger. They don't need to be in records or out of records. They just need to be in scope.

Reply:
So... there seems to be a problem (dark grey background to the definition) if we keep it as an `instance` and define these things in `Core` with the `open _∣_ using (quotient) public` declaration...?